dear_petition/petition/forms.py

```diff
@@ -21,13 +27,34 @@ def save(self, user):
             record.data = record.parse_report(file_)
             if "error" in record.data:
                 raise forms.ValidationError(record.data["error"])
-            if "Defendant" in record.data and "Name" in record.data["Defendant"]:
-                label = record.data["Defendant"]["Name"]
-                record.label = label
-                if idx == 0:
-                    batch.label = label
-                    batch.save()
-            record.save()
+            record.label = record.data.get("Defendant", {}).get("Name", "")
+            if record.label and idx == 0:
+                batch.label = record.label
+                batch.save()
+            record.refresh_record_from_data()
+            offenses = record.data.get("Offense Record", {})
+            if offenses:
+                offense = Offense(
+                    ciprs_record=record,
+                    disposed_on=offenses.get("Disposed On", None),
+                    disposition_method=offenses.get("Disposition Method", ""),
+                )
+                offense.save()
+                offense_records = offenses.get("Records", [])
+                for offense_record in offense_records:
+                    try:
+                        code = int(offense_record.get("Code"))
+                    except ValueError:
+                        code = None
+                    o_record = OffenseRecord(
+                        offense=offense,
+                        law=offense_record.get("Law", ""),
+                        code=code if offense_record.get("Code", "") != "" else None,
```

Comment:
The try/except clause above handles the case that there is no code, so this can be just `code=code,`

Reply:
done